File: Euclid.Storage/Database/Access/CatalogueDao.cs
using Euclid.Storage.Database.Data;
using NHibernate.Linq;
using System.Collections.Generic;
using System.Linq;

namespace Euclid.Storage.Database.Access
{
    public class CatalogueDao
    {
        /// <summary>
        /// Get list of all definition data
        /// </summary>
        public static List<CatalogueData> GetCatalogueOffers()
        {
            using (var session = SessionFactoryBuilder.Instance.SessionFactory.OpenSession())
            {
                return session.QueryOver<CatalogueData>().List() as List<CatalogueData>;
            }
        }

    }
}

File: euclid-storage/src/main/java/euclid/storage/database/access/CatalogueDao.java
package euclid.storage.database.access;

import euclid.storage.database.SessionFactoryBuilder;
import euclid.storage.database.data.CatalogueData;
import org.hibernate.Session;
import org.hibernate.query.Query;

import java.util.List;

public class CatalogueDao {

    public static List<CatalogueData> getCatalogueOffers() {
        try (Session session = SessionFactoryBuilder.getInstance().getSessionFactory().openSession()) {
            Query<CatalogueData> q = session.createQuery("FROM CatalogueData", CatalogueData.class);
            return q.getResultList();
        }
    }
}
